applications/FluidDynamicsApplication/custom_utilities/fluid_auxiliary_utilities.cpp

```diff
@@ -0,0 +1,222 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//                   Suneth Warnakulasuriya
+//
+
+// System includes
+
+
+// External includes
+
+
+// Project includes
+#include "modified_shape_functions/triangle_2d_3_modified_shape_functions.h"
+#include "modified_shape_functions/tetrahedra_3d_4_modified_shape_functions.h"
+#include "utilities/parallel_utilities.h"
+
+// Application includes
+#include "fluid_auxiliary_utilities.h"
+
+namespace Kratos
+{
+
+bool FluidAuxiliaryUtilities::IsSplit(const Vector& rNodalDistances)
+{
+    std::size_t n_pos(0);
+    std::size_t n_neg(0);
+    const std::size_t pts_number = rNodalDistances.size();
+    for (std::size_t i_node = 0; i_node < pts_number; ++i_node){
+        if (rNodalDistances[i_node] > 0.0)
+            n_pos++;
+        else
+            n_neg++;
+    }
+    return (n_pos > 0 && n_neg > 0) ? true : false;
+}
+
+bool FluidAuxiliaryUtilities::IsPositive(const Vector& rNodalDistances)
+{
+    std::size_t n_pos (0);
+    const std::size_t pts_number = rNodalDistances.size();
+    for (std::size_t i_node = 0; i_node < pts_number; ++i_node){
+        if (rNodalDistances[i_node] > 0.0)
+            n_pos++;
+    }
+    return (n_pos == pts_number) ? true : false;
+}
+
+bool FluidAuxiliaryUtilities::IsNegative(const Vector& rNodalDistances)
+{
+    return !IsPositive(rNodalDistances);
+}
+
+double FluidAuxiliaryUtilities::CalculateFluidVolume(const ModelPart& rModelPart)
+{
+    KRATOS_ERROR_IF(rModelPart.NumberOfElements() == 0) << "There are no elements in the provided model part. Fluid volume cannot be computed." << std::endl;
```

Comment:
this will crash in MPI, you need to check the global number

```suggestion
    KRATOS_ERROR_IF(rModelPart.GetCommunicator().GlobalNumberOfElements() == 0) << "There are no elements in the provided model part. Fluid volume cannot be computed." << std::endl;
```

Reply:
Agree. Thanks for spotting!